File: src/main/java/cc/moecraft/logger/coloring/GradientPresets.java
package cc.moecraft.logger.coloring;

import java.awt.*;

import static cc.moecraft.logger.coloring.MultiPointLinearGradient.GradientPoint;

/**
 * 此类由 Hykilpikonna 在 18-7-7 创建!
 * Created by Hykilpikonna on 18-7-7!
 * Twitter: @Hykilpikonna
 * QQ/Wechat: 871674895
 */
public class GradientPresets
{
    public static final MultiPointLinearGradient
            RAINBOW, // 彩虹
            BOP,     // 蓝橙粉
            BPR      // 蓝紫红
    ;

    static
    {
        RAINBOW = new MultiPointLinearGradient(
                new Color(255, 0,   0  ),
                new GradientPoint(new Color(255, 0,   255), 15),
                new GradientPoint(new Color(0,   0,   255), 19),
                new GradientPoint(new Color(0,   255, 255), 15),
                new GradientPoint(new Color(0,   255, 0  ), 18),
                new GradientPoint(new Color(255, 255, 0  ), 17),
                new GradientPoint(new Color(255, 0,   0  ), 16)
        );

        BOP = new MultiPointLinearGradient(
                new Color(64, 224, 208),
                new Color(255, 140, 0),
                new Color(255, 0, 128)
        );

        BPR = new MultiPointLinearGradient(
                new Color(18, 194, 233),
                new Color(196, 113, 237),
                new Color(246, 79, 89)
        );
    }
}

File: src/main/kotlin/cc/moecraft/logger/coloring/GradientPresets.kt
package cc.moecraft.logger.coloring

import cc.moecraft.logger.coloring.MultiPointLinearGradient.GradientPoint
import java.awt.Color

object GradientPresets
{
    var RAINBOW: MultiPointLinearGradient? = null

    // 彩虹
    var BOP: MultiPointLinearGradient? = null

    // 蓝橙粉
    var BPR // 蓝紫红
        : MultiPointLinearGradient? = null

    init
    {
        RAINBOW = MultiPointLinearGradient(
            Color(255, 0, 0),
            GradientPoint(Color(255, 0, 255), 15),
            GradientPoint(Color(0, 0, 255), 19),
            GradientPoint(Color(0, 255, 255), 15),
            GradientPoint(Color(0, 255, 0), 18),
            GradientPoint(Color(255, 255, 0), 17),
            GradientPoint(Color(255, 0, 0), 16)
        )
        BOP = MultiPointLinearGradient(
            Color(64, 224, 208),
            Color(255, 140, 0),
            Color(255, 0, 128)
        )
        BPR = MultiPointLinearGradient(
            Color(18, 194, 233),
            Color(196, 113, 237),
            Color(246, 79, 89)
        )
    }
}
